src/Uno.UI/UI/Xaml/Shapes/Shape.cs

```diff
@@ -25,99 +30,123 @@ private protected double ActualStrokeThickness
 		}
 
 		#region Fill Dependency Property
-
 		//This field is never accessed. It just exists to create a reference, because the DP causes issues with ImageBrush of the backing bitmap being prematurely garbage-collected. (Bug with ConditionalWeakTable? https://bugzilla.xamarin.com/show_bug.cgi?id=21620)
 		private Brush _fillStrongref;
 		public Brush Fill
 		{
-			get { return (Brush)this.GetValue(FillProperty); }
+			get => (Brush)this.GetValue(FillProperty);
 			set
 			{
 				this.SetValue(FillProperty, value);
 				_fillStrongref = value;
 			}
 		}
 
-		public static readonly DependencyProperty FillProperty =
-			DependencyProperty.Register(
-				"Fill",
-				typeof(Brush),
-				typeof(Shape),
-				new FrameworkPropertyMetadata(
-					defaultValue: SolidColorBrushHelper.Transparent,
-					options: FrameworkPropertyMetadataOptions.ValueInheritsDataContext,
-					propertyChangedCallback: (s, e) => ((Shape)s).OnFillChanged((Brush)e.NewValue)
-				)
-			);
-
+		public static DependencyProperty FillProperty { get; } = DependencyProperty.Register(
+			"Fill",
+			typeof(Brush),
+			typeof(Shape),
+			new FrameworkPropertyMetadata(
+				defaultValue: SolidColorBrushHelper.Transparent,
+#if LEGACY_SHAPE_MEASURE
+				options: FrameworkPropertyMetadataOptions.ValueInheritsDataContext,
+				propertyChangedCallback: (s, e) => ((Shape)s).OnFillChanged((Brush)e.NewValue)
+#else
+				options: FrameworkPropertyMetadataOptions.ValueInheritsDataContext | FrameworkPropertyMetadataOptions.AffectsArrange,
```

Comment:
AffectsArrange?? Can produce unwanted perf issues because of relayouting.
+ same comment for similar properties...

Reply:
It does affects arrange, there is no real/acceptable way make it more performant ...